### PR TITLE
Add example of maml training on omniglot

### PR DESCRIPTION
**Pull request by zou3519**
This is related to #328. This PR adds an actually correct
implementation of maml to the repo. The previous implementation doesn't
actually compute higher order gradients where it is supposed to.

I'm not familiar with how torchbench works so please let me know if
there are additional files that need to be modified.

Test Plan:

Ran the following:
```
python test.py -k test_maml_omniglot_example_cpu
python test.py -k test_maml_omniglot_eval_cpu
python test.py -k test_maml_omniglot_train_cpu
```

Future work:
- Delete the maml example that is currently in this repo (or rename it
to make it clear that it's doing something different from the paper that
it is trying to reproduce).

### CODE REVIEW

**Review by Krovatkin — DISMISSED** [review]
:shipit: this is probably the highest density of optimizers per 10 LOCs, I've seen O_o

**Comment by zou3519** [comment]
@Krovatkin It's been a while since you initially reviewed this PR. I rebased on top of the master branch and the build+tests now fully succeed. Could I ask you to take another look at this please before I hit the merge button?

File: torchbenchmark/models/maml_omniglot/__init__.py
Comment (xuzhao9) [inline torchbenchmark/models/maml_omniglot/__init__.py:31]: Please specify a task to the model: https://github.com/pytorch/benchmark/blob/master/torchbenchmark/tasks.py.

File: torchbenchmark/models/maml_omniglot/__init__.py
Comment (xuzhao9) [inline torchbenchmark/models/maml_omniglot/__init__.py:100]: Do we need to set model.eval() or "with torch.no_grad()" in eval()?

File: torchbenchmark/models/maml_omniglot/__init__.py
Comment (zou3519) [reply torchbenchmark/models/maml_omniglot/__init__.py:31]: This model is metalearning over classification on an image dataset. I think this is either OTHER.OTHER_TASKS or COMPUTER_VISION.OTHER_COMPUTER_VISION. Do you have advice for which category this should go @xuzhao9? (Otherwise I'll pick OTHER.OTHER_TASKS)

Comment (xuzhao9) [reply torchbenchmark/models/maml_omniglot/__init__.py:31]: > This model is metalearning over classification on an image dataset. I think this is either OTHER.OTHER_TASKS or COMPUTER_VISION.OTHER_COMPUTER_VISION. Do you have advice for which category this should go @xuzhao9? (Otherwise I'll pick OTHER.OTHER_TASKS)

Since it is meta learning, please add it to OTHER.OTHER_TASKS, thanks!

**Comment by xuzhao9** [comment]
Please help resolve the comments and I will be happy to merge your contribution!

File: torchbenchmark/models/maml_omniglot/__init__.py
Comment (zou3519) [reply torchbenchmark/models/maml_omniglot/__init__.py:103]: Added eval() and no_grad

File: torchbenchmark/models/maml_omniglot/__init__.py
Comment (zou3519) [reply torchbenchmark/models/maml_omniglot/__init__.py:33]: Added OTHER.OTHER_TASKS.

**Comment by zou3519** [comment]
@xuzhao9 I resolved the comments, could you please take another look when you get a chance?

**Review by xuzhao9 — APPROVED** [review]
LGTM! Thanks for the contribution!

**Pull request merged**
